crates/wallet/src/wallet/mod.rs

```diff
@@ -906,6 +877,23 @@ impl Wallet {
         self.stage.append(additions.into());
     }
 
+    /// Inserts an `anchor` for a transaction with the given `txid`.
+    ///
+    /// This stages the changes, you must persist them later.
+    pub fn insert_anchor(&mut self, txid: Txid, anchor: ConfirmationTimeHeightAnchor) {
+        let indexed_graph_changeset = self.indexed_graph.insert_anchor(txid, anchor);
+        self.stage.append(indexed_graph_changeset.into());
+    }
```

Comment:
I feel like we should include a warning that the caller is responsible for ensuring the correctness of the information provided. If the caller anchors a tx in a block where the tx does not actually belong in, the canonical history of the wallet may be messed up permanently (since the `TxGraph` is monotone).